cypress/e2e/chapters/[chapterId].cy.ts

```diff
@@ -8,39 +7,24 @@ describe('chapter page', () => {
     cy.task('seedDb');
   });
 
+  //ToDo: check for if UI components are functioning as intended
   it('user can join chapter and change subscription status', () => {
-    cy.login('test@user.org');
-    cy.visit(`/chapters/${chapterId}`);
-
-    cy.findByRole('button', { name: 'Join chapter' }).click();
-    cy.findByRole('button', { name: 'Confirm' }).click();
-
-    cy.contains(/joined chapter/);
-    cy.contains(/Join chapter/).should('not.exist');
-    cy.contains(/Unsubscribe/);
-
-    cy.findByRole('button', { name: 'Unsubscribe' }).click();
-    cy.findByRole('button', { name: 'Confirm' }).click();
-
-    // TODO Check if user event_reminders were cleared and user event_users unsubscribed for events in this chapter. And other chapters were not affected.
-    cy.contains(/unsubscribed/);
-    cy.contains(/Subscribe/);
-
-    cy.findByRole('button', { name: 'Subscribe' }).click();
-    cy.findByRole('button', { name: 'Confirm' }).click();
-
-    cy.contains(/subscribed/);
-
-    cy.task<ChapterMembers>('getChapterMembers', chapterId).then(
-      (chapter_users) => {
-        expect(
-          chapter_users.findIndex(
-            ({ user: { email }, subscribed }) =>
-              email === 'test@user.org' && subscribed,
-          ),
-        ).to.not.equal(-1);
+    cy.login();
+
+    cy.joinChapter(chapterId, { withAuth: true }).then((response) => {
```

Comment:
This logins as owner - `foo@bar.com`, who is seeded as a member of chapter with id `1`.

Reply:
Owner can leave chapter 🤔🤦

Reply:
I mean, changed code checks if when user being already part of chapter, and tries to join it again, doesn't end up with error.

Reply:
Well, time to update the function to

- stop joined member from joining
- stop owner from leaving
- update cypress tests to check for these

Reply:
- In general I think that's somewhat convenient, to kind of ignore on the server side further attempts to join chapter. I'm not sure if it's worth having it tested explicitly or changing.
- Why instance owner shouldn't be allowed to leave chapter?

Reply:
> In general I think that's somewhat convenient, to kind of ignore on the server side further attempts to join chapter. I'm not sure if it's worth having it tested explicitly or changing.

I have no idea, I like the calls to server to be tested by cypress because I don't have that much experience with dealing with server side issues, so I feel comfortable with safe nets around the server-side.

> Why instance owner shouldn't be allowed to leave chapter?

The word owner confused me, and I thought he was owner of chapter not instance anyway, you are right it doesn't matter 👍 

---

> I don't have that much experience with dealing with server side issues

This does count chapter, which is scary 😨. And I may lack experience in client side as well. 🤷